xds/src/main/java/io/grpc/xds/XdsClient.java

```diff
@@ -59,6 +59,9 @@
     final boolean hasFaultInjection;
     @Nullable // Can be null even if hasFaultInjection is true.
     final HttpFault httpFault;
+    // Server side Listener.
+    @Nullable
+    final Listener listener;
```

Comment:
Can we flatten the `Listener` class? As what C-core is doing: https://github.com/grpc/grpc/blob/0937cb1249129a84159c639c59f49a451157edf7/src/core/ext/xds/xds_api.h#L244-L301

Having a `Listener` blob here is vague, as its meaning is unintuitive. Also, we would want to eliminate `EnvoyServerProtoData`/`EnvoyProtoData` and move the parsing logic into XdsClient's implementation, as that's the only place that code is used. So, no need to have this extra layer of reference.

Reply:
You could do that in a separate step. I won't block you on this.

Reply:
I am doing this as per the plan we discussed. This PR has the limited goal of adding server side processing in `ClientXdsClient` and once the server side has started to use it then doing further changes. At that point we can take up flattening `Listener`